### ISSUE TITLE
bad colours in VS code (compared to rendering in browser)

### ISSUE THREAD

**Issue by tibor-mach** [iterative/vscode-dvc#1957]
with the default VS code dark mode, some plots (e.g. the confusion matrix) are hard to read (see the first picture). They are not much better with other VS code themes, unfortunately. However, they are just fine when opened in a browser (see the second picture)
![image](https://user-images.githubusercontent.com/56956998/176249829-e483eab1-9c04-4951-b470-208f79694c74.png)

![image](https://user-images.githubusercontent.com/56956998/176249846-55f143dd-d726-4f51-9dbe-1df8dee1654b.png)

The browser is Firefox, but it should not matter.

**Comment by shcheklein** [comment]
🤔 Should we put them on the white background then to make readable? Otherwise it feels like we are breaking what's coming our way just to make it look nice?

**Comment by shcheklein** [comment]
I wonder if we can at least make some slight modifications to the existing built-in templates to make them look better of all the backgrounds - @pared @daavoo wdyt?

**Comment by tibor-mach** [comment]
@rogermparent 

But the plots do not look very good in VS code even in the light mode (which I also wonder how many people actually use :-) ), at least with the [AUR version](http://aur.archlinux.org/packages/visual-studio-code-bin) of VS code on Manjaro. In fact it looks to me like the colours are suppose do work better with the dark theme.
![image](https://user-images.githubusercontent.com/56956998/176601481-cc9bca98-9a24-41d1-9bca-12108d8bbfb5.png)

They look fine in the browser (as shown above). So if they could be rendered in VS code with a white background so that they look exactly as in the browser (like @shcheklein suggested) that would probably be the best solution...plus then you do not need any theming and they will be rendered exactly the same for any theme.

**Comment by daavoo** [comment]
> I wonder if we can at least make some slight modifications to the existing built-in templates to make them look better of all the backgrounds - @pared @daavoo wdyt?

I would first need to understand why the text color is being rendered differently in the extension vs in the browser. 
I appears to be some processing done on the extension side before rendering that modifies the vega template (https://github.com/iterative/vscode-dvc/blob/main/extension/src/plots/vega/util.ts) so changes in the built-in templates would still be discarded?

**Comment by daavoo** [comment]
I tried using a custom template where I modify the `text` property of the encoding and the `colorscale`. 
`dvc plots show` correctly displays both changes but the extension ignores the `text` changes.

Note the red text on the diagonal:

<img width="923" alt="Captura de Pantalla 2022-06-30 a las 9 41 53" src="https://user-images.githubusercontent.com/12677733/176621312-4c8739b3-b59b-4715-8b89-ceada2484334.png">

Missing in the extension:

<img width="862" alt="Captura de Pantalla 2022-06-30 a las 9 42 44" src="https://user-images.githubusercontent.com/12677733/176621483-2dd4ab1c-3d2a-4283-9257-43208ea912fb.png">

**Comment by arcticbear** [comment]
@daavoo I think this happens because `stroke="#d4d4d4"` overrides all the text label colors, as I mentioned above.

**Comment by daavoo** [comment]
> @daavoo I think this happens because stroke="#d4d4d4" overrides all the text label colors, as I mentioned above.

Do you know how/where is that inserted in the code?

**Comment by arcticbear** [comment]
Nope, maybe @rogermparent knows?

**Comment by pared** [comment]
> I wonder if we can at least make some slight modifications to the existing built-in templates to make them look better of all the backgrounds 

@shcheklein its a matter of adding a single field to the template. Also we would need to choose the best color range:
https://vega.github.io/vega/docs/schemes/

**Comment by rogermparent** [comment]
Sorry for the late response! I'm working on this right now and seem to have broken through with a solution inspired by Vega lite's theme API.

> But the plots do not look very good in VS code even in the light mode

 I discovered a tangentially related issue in where the main color is calculated statically from the CSS variable on load instead of just using the CSS variable, which means it doesn't switch with VSCode's theme. Replacing this calculation with the variable makes light/dark adapt, though it doesn't fix the whole issue as there can still be contrast conflicts within elements with colored backgrounds like confusion matrices.

Basically, if we override less aggressively we can make outer labels adapt to light/dark while not mangling inner text. Here's some screenshots from #1988

![image](https://user-images.githubusercontent.com/9111807/177426220-9a632e9a-18a3-4fea-b0a2-1980b47b6c76.png)
![image](https://user-images.githubusercontent.com/9111807/177426281-e0385cb4-b258-4bc8-9c0c-b8e063385156.png)

Note how the confusion matrices now include the original dual-toned colors depending on the cell backgrounds.

**Comment by tibor-mach** [comment]
@rogermparent thanks! This looks much better
